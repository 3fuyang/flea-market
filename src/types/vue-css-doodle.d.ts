declare module '@luxdamore/vue-css-doodle'