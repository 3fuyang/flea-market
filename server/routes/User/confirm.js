// Confirm 页面
const express = require('express')
const { result } = require('lodash')
const app = express()

const connection = require('../../database/db')

app.use(express.json())
app.use(express.urlencoded({extended:  false}))

// 获取买家姓名、手机号
app.get(`/getBuyerInfo/:user_id`, (req, res) => {
  connection.query(
    `select real_name,telnum from userAccount where user_id='${req.params.user_id}'`,
    (err, result) => {
      if (err) throw err
      res.end(JSON.stringify(result))
    }
  )
})

// 获取商品信息
app.post(`/goodsToConfirm`, (req, res) => {
  const promises = []
  req.body.forEach(item => {
    promises.push(new Promise((resolve) => {
      connection.query(
        `select good_id,title,price,images from goodInfo where good_id='${item}';
         select nickname from userAccount where user_id=(select seller_id from goodInfo where good_id='${item}')`,
        (err, result) => {
          if (err) throw err
          let raw = JSON.parse(JSON.stringify(result)).flat(2)
          resolve(Object.assign(raw[0], raw[1]))
        }
      )
    }))
  })
  Promise.all(promises)
    .then(result => {
      res.end(JSON.stringify(result))
    })
})

module.exports = app