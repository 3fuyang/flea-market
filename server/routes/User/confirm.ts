// Confirm 页面
import express from 'express'
import connection from '../../database/db'

const app = express()

app.use(express.json())
app.use(express.urlencoded({extended:  false}))

// 获取买家姓名、手机号
app.get(`/getBuyerInfo/:user_id`, (req, res) => {
  connection.query(
    `select real_name,telnum from userAccount where user_id=?`,
    [req.params.user_id],
    (err, result) => {
      if (err) throw err
      res.end(JSON.stringify(result))
    }
  )
})

// 获取商品信息
app.post(`/goodsToConfirm`, (req, res) => {
  const promises: any[] = []
  req.body.forEach((item: any) => {
    promises.push(new Promise((resolve) => {
      connection.query(
        `select good_id,title,price,images,seller_id from goodInfo where good_id=? and available=0;
         select nickname from userAccount where user_id=(select seller_id from goodInfo where good_id=? and available=0)`,
         [item, item],
        (err, result) => {
          if (err) throw err
          let raw = JSON.parse(JSON.stringify(result)).flat(2)
          resolve(Object.assign(raw[0], raw[1]))
        }
      )
    }))
  })
  Promise.all(promises)
    .then(result => {
      res.end(JSON.stringify(result))
    })
})

// 生成订单
app.post(`/generateOrder`, (req, res) => {
  const {buyer, seller, goodID, price, generatedTime, stat} = req.body
  // 向orderData插入新订单，在goodInfo中更新商品为不可访问，从买家的shoppingCart中删除商品
  connection.query(
    `insert into orderData(buyer, seller, good_id, price, generated_time, stat) values (?, ?, ?, ?, ?, ?);
     update goodInfo set available=1 where good_id=?;
     delete from shoppingCart where good_id=? and user_id=?`,
     [buyer, seller, goodID, price, generatedTime, stat, goodID, goodID, buyer],
    (err, result) => {
      if (err) throw err
      res.end(JSON.stringify(result))
    }
  )
})

export default app