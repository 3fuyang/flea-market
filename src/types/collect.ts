export class Collect {
  userID: string = ''
  goodID: string = ''
  collected: boolean = false
  collect () {
    if (!this.userID) {
      return '请先登录'
    } else if (this.userID.length === 4) {
      return '管理员请登录普通账号进行收藏'
    } else if (!this.goodID) {
      return '商品不存在'
    } else if (this.collected) {
      return '不能重复收藏'
    } else {
      return true
    }
  }
  set (uid: string = '', gid: string = '', collected: boolean = true) {
    this.userID = uid
    this.goodID = gid
    this.collected = collected
  }
}