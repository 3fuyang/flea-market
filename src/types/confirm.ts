export class Confirm {
  orderID: string = '1'
  stat: '待付款' | '待确认' | '待评价' | '已完成' = '待确认'
  confirm () {
    switch (this.stat) {
      case '待付款':
        return '请先付款'
      case '待确认':
        return true
      default:
        return '不可重复确认'
    }
  }
  set (st: '待付款' | '待确认' | '待评价' | '已完成' = '待确认') {
    this.stat = st
  }
}