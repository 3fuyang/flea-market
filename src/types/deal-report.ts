export class DealReport {
  adminID: string = '0001'
  orderID: string = '1'
  reported: '待处理' | '已封禁' | '已驳回' = '待处理'
  reply: string = ''
  dealReport (stat: '待处理' | '已封禁' | '已驳回') {
    if (this.reported !== '待处理') {
      return '不可重复处理举报'
    } else if (!this.reply) {
      return '请输入回复'
    } else {
      this.reported = stat
      return true
    }
  }
  set (rptd: '待处理' | '已封禁' | '已驳回', rpl: string) {
    this.reported = rptd
    this.reply = rpl
  }
}