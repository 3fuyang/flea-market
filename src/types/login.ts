// 测试用，类型建模
export class LoginForm {
  // 类型
  type: 'member' | 'admin' = 'member'
  // 用户名
  userID: string = ''
  // 密码
  password: string = ''
  // 登录
  login () {
    if (!this.userID) {
      return '请输入账号'
    } else if (this.type === 'member' && this.userID.length !== 7) {
      return '普通用户账号应长7位'
    } else if (this.type === 'admin' && this.userID.length !== 4) {
      return '管理员账号应长4位'
    } else if (!this.password) {
      return '请输入密码'
    } else {
      // 登录
      return true
    }
  }
  // setter
  set (tp: 'member' | 'admin' = 'member', id: string = '', pwd: string = '') {
    this.type = tp
    this.userID = id
    this.password = pwd
  }
}