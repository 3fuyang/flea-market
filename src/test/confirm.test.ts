import { describe, it, expect } from 'vitest'
import { Confirm } from '@/types/confirm'

const confirm = new Confirm()

let counter = 1

const test = (stat: '待付款' | '待确认' | '待评价' | '已完成', expc: string | boolean) => {
  return it(`cf_${counter++}`, () => {
    confirm.set(stat)
    expect(confirm.confirm()).toBe(expc)
    expect(confirm).toMatchSnapshot()
  })
}

describe('confirm', () => {
  test('待付款', '请先付款')
  test('待确认', true)
  test('待评价', '不可重复确认')
})