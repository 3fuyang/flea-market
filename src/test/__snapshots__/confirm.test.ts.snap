// Vitest Snapshot v1

exports[`confirm > cf_1 1`] = `
Confirm {
  "orderID": "1",
  "stat": "待付款",
}
`;

exports[`confirm > cf_2 1`] = `
Confirm {
  "orderID": "1",
  "stat": "待确认",
}
`;

exports[`confirm > cf_3 1`] = `
Confirm {
  "orderID": "1",
  "stat": "待评价",
}
`;
